lib/Raven/Serializer.php

```diff
@@ -109,6 +109,12 @@ protected function serializeValue($value)
         if (is_null($value) || is_bool($value) || is_float($value) || is_integer($value)) {
             return $value;
         } elseif (is_object($value) || gettype($value) == 'object') {
+            // If the value is a callable and expects no parameters, we evaluate the
+            // callable value by calling it.
+            if (is_callable($value) && Raven_Util::getCallableParamNum($value) === 0) {
+                return $value();
```

Comment:
>pardon my asking, but as a non-native English speaker

I'm not native English speaker either 😄 

Btw, what I meant is that functions could do some heavy work inside, for example they could do heavy operations to compute some value or executes some queries that take time, so we shouldn't call indiscriminately all of them just to serialize the result value. Plus, maybe an user wants to serialize just some values: if before nothing was serialized at all, with your changes everything is serialized now. My suggestion, which anyway is beyond the original topic of this PR and doesn't address your problem, is to serialize the name, argument types and return type of a closure, but not the return value itself. I understand why you suggested to call the closures and I'm not sure this is enough, but you could use a custom processor to add data on-the-fly to the event being sent to the server